recovery_reader.go

```diff
@@ -0,0 +1,70 @@
+package nutsdb
+
+import (
+	"bufio"
+	"encoding/binary"
+	"os"
+)
+
+// fileRecovery use bufio.Reader to read entry
+type fileRecovery struct {
+	reader *bufio.Reader
+}
+
+func newFileRecovery(path string) (fr *fileRecovery, err error) {
+	fd, err := os.OpenFile(path, os.O_RDWR, os.ModePerm)
+	if err != nil {
+		return nil, err
+	}
+	return &fileRecovery{
+		reader: bufio.NewReader(fd),
+	}, nil
+}
+
+func (fr *fileRecovery) readEntry() (e *Entry, err error) {
```

Comment:
Recovery is easy to be associated with recovery related to golang's panic. Is it inappropriate for this file name?
 In addition, this readEntry method is very similar to the readAt code in the datafile (a lot of codes are the same https://github.com/nutsdb/nutsdb/blob/master/datafile.go#L60)

Reply:
fileRecovery means recover file to entries in memory, and the code of readEntry method is very similar to the readAt code in the datafile, its only change the interact between nutsdb and file.

Reply:
I understand what you mean, but don't repeat the code, can it be refactored?

Reply:
have a little refactor, please review it when you have time.

Reply:
ok